causallearn/utils/KCI/KCI.py

```diff
@@ -145,8 +145,16 @@ def kernel_matrix(self, data_x, data_y):
         else:
             raise Exception('Undefined kernel function')
 
-        data_x = stats.zscore(data_x, axis=0)
-        data_y = stats.zscore(data_y, axis=0)
+        if np.var(data_x) == 0:
+            data_x -= np.average(data_x)
+        else:
+            data_x = stats.zscore(data_x, axis=0)
```

Comment:
+ Current `np.var(data_x)` does not support multi-dim `data_x` with some dims being constant:
```python
In [7]: data_x = np.random.randn(100, 1)

In [8]: data_y = np.ones_like(data_x) # constant

In [9]: data_xy = np.hstack([data_x, data_y])

In [10]: stats.zscore(data_xy, axis=0)
Out[10]: 
array([[ 4.15513535e-01,             nan],
       [-1.71903423e+00,             nan],
       [ 7.59493517e-01,             nan],
       [-1.34182046e+00,             nan],
       ...
```

+ For numpy floating points, how to reliably identify a constant array? `var=0` is not enough:
```python
In [16]: arr1 = np.array([1., 1., 1.])

In [17]: np.var(arr1)
Out[17]: 0.0

In [18]: stats.zscore(arr1)
Out[18]: array([nan, nan, nan])

#########

In [19]: arr2 = np.array([-0.087, -0.087, -0.087])

In [20]: np.var(arr2)
Out[20]: 1.925929944387236e-34

In [21]: np.var(arr2) == 0
Out[21]: False

In [22]: stats.zscore(arr2)
Out[22]: array([nan, nan, nan]) # though np.var != 0, here it still returns nan
```

+ Based on above, can we just mask nan values to zero after `stats.zscore`? `data_x[np.isnan(data_x)] = 0.`

Reply:
> This operation is safe, since we would (expect to) check any raw data_x (before kci) so it does not contain nan values, the nan returned in the normalized array is only due to constant (not original nan values).

Do we ensure this in code, i.e. ensure raw data_x doesn't contain nan value?

Reply:
> > This operation is safe, since we would (expect to) check any raw data_x (before kci) so it does not contain nan values, the nan returned in the normalized array is only due to constant (not original nan values).
> 
> Do we ensure this in code, i.e. ensure raw data_x doesn't contain nan value?

Not yet. I'll do this soon after this pr is merged.

Reply:
The problem is already fixed. I guess maybe we can merge now~ 

Many thanks to all of your(@kunwuz, @MarkDana, @tofuwen ) help~ Since I am at the first time contributing to a community codebase and lack experience.

> Thanks so much for your awesome contributions, @cogito233 , @MarkDana and @tofuwen ! @cogito233 Please let us know if you think the current PR is ready to go so we will solve the remaining issues in a new PR. Or we could include these in this PR if you would like to. :-)

Reply:
Thanks so much! @tofuwen @MarkDana When you have time, could you please make a final check on the current PR and let me know when it's ready to be merged? Many thanks!